src/lib/routers/history.js

```diff
@@ -15,6 +15,12 @@ function defaultParseURL({ qsModule, location }) {
   return qsModule.parse(location.search.slice(1));
 }
 
+function setWindowTitle(title) {
+  if (title) {
+    window.document.title = title;
```

Comment:
we aren't checking that these variables are available, however that's the same in other parts of this module too

Reply:
You mean `window`?

Reply:
yes, if window is not defined (i.e. in server side rendering) this code will throw

Reply:
We've got at least two more occurrences of `window` without check:

- https://github.com/algolia/instantsearch.js/blob/577e4ccb33155d0d4678b04505132d39a2918f22/src/connectors/numeric-menu/connectNumericMenu.js#L133
- https://github.com/algolia/instantsearch.js/blob/577e4ccb33155d0d4678b04505132d39a2918f22/src/widgets/analytics/analytics.js#L176

We can handle those when we plan to be SSR-ready.